sdk/identity/azure-identity-v2/src/main/java/com.azure.identity.v2/implementation/IdentityClientOptions.java

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.v2.implementation;
+
+import com.azure.identity.v2.AzureAuthorityHosts;
+import com.azure.identity.v2.TokenCachePersistenceOptions;
+import com.azure.identity.v2.implementation.util.ValidationUtil;
+import io.clientcore.core.http.client.HttpClient;
+import io.clientcore.core.http.pipeline.HttpInstrumentationOptions;
+import io.clientcore.core.http.pipeline.HttpPipeline;
+import io.clientcore.core.http.pipeline.HttpPipelinePolicy;
+import io.clientcore.core.http.pipeline.HttpRetryOptions;
+import io.clientcore.core.http.pipeline.HttpRedirectOptions;
+import io.clientcore.core.instrumentation.logging.ClientLogger;
+import io.clientcore.core.utils.configuration.Configuration;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Options to configure the IdentityClient.
+ */
+public class IdentityClientOptions implements Cloneable {
```

Comment:
```suggestion
public class CredentialOptions implements Cloneable {
```

or `IdentityOptions` maybe? Just need to get `Client` out of there.

Reply:
This is refactored to cater to Confidential / Public / Managed identity options separately, will help to keep clear distinction in flows and make it easier to maintain.